modules/friendlyshared.js

```diff
@@ -144,7 +150,7 @@ Twinkle.shared.callbacks = {
 			return;
 		}
 
-		Morebits.status.info( 'Info', 'Will add the shared IP address template to the top of the user\'s talk page.' );
+		Morebits.status.info( $.i18n._('tw-core-morebits-info'), $.i18n._('tw-sharedip-will-add') );
 		text += params.value + '|' + params.organization;
 		if( params.value === 'Shared IP edu' && params.contact !== '') {
```

Comment:
As I said above, I'm not asking for a perfect solution - leaving the callbacks in is, for now, OK. However, I really have a hard time swallowing the idea of keeping some hard-coded template names in. 

> Besides, this is something I have raised over and over again: that EN WP logic is embedded in Twinkle. And that is bad, we all know, but it is not l10n/i18n's job to fix it. 

In principle, I disagree. To me, localising Twinkle is precisely removing all enwiki-specific strings, objects, logic etc. from the core code and into a separate localisation file. However, it certainly does not have to be done all at once, and as you say, it is a Herculean effort (which is why it hasn't been done until now). Some localisation is better than none, so I'm prepared to accept a partially-completed transition.

I know I frustrate you by my constant conflation of i18n and l10n, but from a Twinkle perspective the overlap is much greater than for most other projects. If I talk about i18n as the translation of project-independent strings and l10n as adaptation to an individual project, then in fact a lot of string translation will fall under the l10n heading.